Core/src/Surfaces/DiscSurface.cpp

```diff
@@ -341,10 +341,10 @@ Acts::Vector3 Acts::DiscSurface::normal(const GeometryContext& gctx,
 
 Acts::Vector3 Acts::DiscSurface::binningPosition(const GeometryContext& gctx,
                                                  BinningValue bValue) const {
-  if (bValue == binR) {
+  if (bValue == binR || bValue == binPhi) {
     double r = m_bounds->binningValueR();
     double phi = m_bounds->binningValuePhi();
-    return Vector3(r * cos(phi), r * sin(phi), center(gctx).z());
+    return localToGlobal(gctx, Vector2{r, phi}, Vector3{});
```

Comment:
Should we generalise this change to all surfaces ? (for ex CylinderSurface uses : `return Vector3(sfCenter.x() + R * cos(phi), sfCenter.y() + R * sin(phi), sfCenter.z());`)

Reply:
I'm not even sure I understand what CylinderSurface wants to calculate there.